azurerm/resource_arm_automation_dsc_nodeconfiguration.go

```diff
@@ -0,0 +1,156 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/automation/mgmt/2015-10-31/automation"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationDscNodeConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationDscNodeConfigurationCreateUpdate,
+		Read:   resourceArmAutomationDscNodeConfigurationRead,
+		Update: resourceArmAutomationDscNodeConfigurationCreateUpdate,
+		Delete: resourceArmAutomationDscNodeConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"automation_account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"content_embedded": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"configuration_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmAutomationDscNodeConfigurationCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationDscNodeConfigurationClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Dsc Node Configuration creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	accName := d.Get("automation_account_name").(string)
+	content := d.Get("content_embedded").(string)
+
+	// configuration name is always the first part of the dsc node configuration
+	// e.g. webserver.prod or webserver.local will be associated to the dsc configuration webserver
+
+	configurationName := strings.Split(name, ".")[0]
+
+	parameters := automation.DscNodeConfigurationCreateOrUpdateParameters{
+		Source: &automation.ContentSource{
+			Type:  automation.EmbeddedContent,
+			Value: &content,
+		},
+		Configuration: &automation.DscConfigurationAssociationProperty{
+			Name: &configurationName,
+		},
+		Name: &name,
```

Comment:
It will fail [validation](https://github.com/Azure/azure-sdk-for-go/blob/master/services/automation/mgmt/2015-10-31/automation/dscnodeconfiguration.go#L62).